### DISCUSSION TITLE
Custom Heirloom

### DISCUSSION THREAD

**Discussion by virgo77** [azerothcore/azerothcore-wotlk#18351] [General]
Hi all, how are you? Hope fine :)

I need some help from "specialists" about Heirloom creation. I explain:
- I found [this script](https://emudevs.gg/resources/custom-heirloom-set-for-azerothcore.321/) in which there's entire 16 pieces heirloom stuff for each class (cloth / leather / mail / plate)
- The Heirloom works when on the character
- But there are some little bugs

When using the `.additem` command, the piece is going correctly into the bag. But:
- the icon is a big red '?'. But when on the character, the icon is the good one
- there's no sound when moving it in an another bag stack or moving it to my character to wear it
- if I right-click on the piece from the bag to wear it, there's an error about the fact I don't meet "proficiency". But if I move the piece on the character, he's wearing it correctly

About the core:
- when launching the core, I got errors saying that the item xxxxx is not present in 'item.dbc'
- but, other that, the core seems working fine

My ideas and solutions:
- about the 'item.dbc' error, I think I found how to fix it by creating entries in the "world/item_dbc" table, with a specific SQL file (or adding each ones manually works too). And the errors when launching the core are now gone away
- but, for the rest, I think I'll have to create a specific .MPQ file, in which I must put the custom Heirloom creations (skins, sounds, item custom number (and maybe other things). I'm using the Windows MPQEditor to access the genuine MPQ files. I tried some search inside each ones to find the skins or sound (with the help of WowHead DB) but I don't find anything. For example: item [48685 - Polished Breastplate of Valor](https://www.wowhead.com/wotlk/item=48685/polished-breastplate-of-valor) is using this Icon: [ inv_chest_plate03](https://www.wowhead.com/wotlk/icon=132738/inv-chest-plate03) . The problem is: where to find it? And same with the sounds, etc. I don't find these files (maybe .blp).

So, if some guys (and girls!) here can help me about how to fix this, I would be glad :)
I'm sure it's not as easy as I think, but I think I'm on the good way.

Thanks a lot for your help ^^

See you!

V.

NB.: I plan to create a Custom NPC which will sell these Custom Heirlooms. And I would like this NPC to be as [Dame Evniki Kapsalis](https://www.wowhead.com/wotlk/npc=34885/dame-evniki-kapsalis) which is unlocked after some quests and reputations. I'm searching about solutions, but don't know how to do this. I will ask for help later :)

**Comment by virgo77** [comment]
Hey There! :)

So, I tried my first idea, which was to create a custom MPQ file:
- I found [this topic](https://maxcheaters.com/topic/83284-tutorial-how-to-fix-custom-items-displayid-icon-and-how-to-fix-custom-weapon/) and test a Custom MPQ file, but it's not working;
- I added my custom heirloom inside the "world/item_dbc" table,  but it's not working;
- I tried both (Custom MPQ File and table modification), but it's not working;

Note that each time I cleaned the Client Cache!

I'm stuck. I don't know what to do else :(

Maybe adding the items in the core compilation?

Thanks for your replies

V.


------------------------

EDIT 1

I moved the patch from the DATA\ENUS folder, as it was told on the topic to the DATA folder, cleaned the cache and... it's working, yeah!

The red "?" disappeared. The sounds are working, and I can right-click to put the stuff on my character.

So this part is OK.

NEXT 1st STEP: creating some specific heirlooms weapons (like crossbow, polearm, 2H Axe).
NEXT 2nd STEP: creating the Custom NPC, like Dame Evniki Kapsalis, to sell these custom heirlooms. No idea if It will be ok.

------------------------

**Comment by Scylune** [comment]
Hey :) 
Did you achieve it ? Does it work ?
Thanks ^^
